src/main/java/org/broadinstitute/hellbender/utils/bwa/Alignment.java

```diff
@@ -0,0 +1,142 @@
+package org.broadinstitute.hellbender.utils.bwa;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.QualityUtil;
+import htsjdk.samtools.util.SequenceUtil;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.utils.BaseUtils;
+import org.broadinstitute.hellbender.utils.QualityUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.read.SAMRecordToGATKReadAdapter;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class Alignment {
```

Comment:
Yes.  I punted on this.  What you'd ideally want to do is to prepare a SAMFileHeader in which the the @SQ tags match the reference against which we aligned.  I can't figure out how to get that info.  (The index information is incomplete.)
Next best might be to check that the reference contig names in the BWA index match the ones in the SAMFileHeader.  Then we might guess that we have a compatible coordinate system.  But then I still don't know where to get complete @SQ info in the case of an unaligned BAM, which won't have @SQ records in its header.
This is a bit of a mess.
For now, I'll just document it as you suggest.